### ISSUE TITLE
Missing parameter of beta

### ISSUE THREAD

**Issue by Corleone-Huang** [lucidrains/vector-quantize-pytorch#22]
Hi, in the original VQVAE paper, the commit_loss is defined as 

```
(quantize.detach()-x) ** 2 + beta * (quantize - x.detach() ** 2)
```

where the beta is usually to be 0.25. But the commit_loss is defined as the following in your implementation:

```
F.mse_loss(quantize.detach(), x)
```

So I wonder if the parameter beta is set to be 1 by default or if the second term is missing? Thank you very much.

**Comment by lucidrains** [comment]
@Corleone-Huang Hello Corleone! So the `beta` would correspond to the `commit_loss_weight` https://github.com/lucidrains/vector-quantize-pytorch/blob/master/vector_quantize_pytorch/vector_quantize_pytorch.py#L543

**Comment by dortenenboim314** [comment]
Hi @lucidrains, seems that the [commitment loss in your code](https://github.com/lucidrains/vector-quantize-pytorch/blob/master/vector_quantize_pytorch/vector_quantize_pytorch.py#L588) refers only to the first term of the equation:

`(quantize.detach()-x) ** 2 + beta * (quantize - x.detach() ** 2)`

Does the second term appears in you code somewhere? If it does, can you please help me find it? I tried and found no references to it.

If it doesn't - what "moves" the **codebook** towards the encoder's output? The commit loss seems to "move" only the encoder outputs.

**Comment by naitian** [comment]
In this repository, the codebook is updated based on exponential moving averages.
